src/coreclr/src/tools/r2rdump/R2RDiff.cs

```diff
@@ -125,6 +125,7 @@ private void ShowDiff(Dictionary<string, int> leftObjects, Dictionary<string, in
 
             int leftTotal = 0;
             int rightTotal = 0;
+            StringBuilder line = new StringBuilder();
```

Comment:
FWIW, this change is not strictly better.  If allKeys is empty, this is going to incur more allocation than it did previously.

Reply:
Agreed, while it can be done as you suggested in EventLog.cs to remove the possible allocation I am not sure if it is worth the extra if/else block for non-libraries code so I am removing the changes on all tools files